lib/cloudant.js

```diff
@@ -745,7 +745,9 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   var id = data[idName];
   if (id) {
     self.getCurrentRevision(model, id, function(err, rev) {
-      if (err) return cb(err);
+      if (err && err.statusCode !== 404) {
+        return cb(err);
+      }
```

Comment:
`updateOrCreate` seems to strongly imply that it'll update the record (not in the Cloudant sense!) or create it. `updateAttributes` is the only "true" update we have in that sense, is it not?